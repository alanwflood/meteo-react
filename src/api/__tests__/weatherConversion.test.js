import mockWeatherResponse from "./mockWeatherResponse.json";
import convertWeatherData from "../weatherConversion";

describe("weatherConversion", () => {
  it("Converts json from a a 5 day weather api call to the correct shape", () => {
    global.Date = class extends Date {
      constructor(...args) {
        if (args.length > 0) {
          // eslint-disable-next-line constructor-super, no-constructor-return
          return super(...args);
        }
        // eslint-disable-next-line no-constructor-return
        return new Date("2020-04-16T11:00:00Z");
      }
    };

    const subject = convertWeatherData;
    expect(subject(mockWeatherResponse)).toMatchSnapshot();
  });
});
