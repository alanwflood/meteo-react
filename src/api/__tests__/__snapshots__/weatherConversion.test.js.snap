// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`weatherConversion Converts json from a a 5 day weather api call to the correct shape 1`] = `
Object {
  "data": Object {
    "1586905200": Object {
      "avgWeather": Object {
        "description": "broken clouds",
        "icon": "04d",
      },
      "descriptions": Array [
        "broken clouds",
        "clear sky",
      ],
      "displayName": null,
      "humidity": Object {
        "avg": "69.00%",
        "data": Array [
          67,
          71,
        ],
      },
      "icons": Array [
        "04d",
        "01n",
      ],
      "rain": Object {
        "avg": "0.00mm",
        "data": Array [
          0,
          0,
        ],
      },
      "snow": Object {
        "avg": "0.00mm",
        "data": Array [
          0,
          0,
        ],
      },
      "temps": Object {
        "avg": "9°C",
        "data": Array [
          11,
          8,
        ],
        "max": "11°C",
        "min": "8°C",
      },
      "times": Array [
        1586973600,
        1586984400,
      ],
      "wind": Object {
        "avg": "2.50m/s",
        "data": Array [
          2,
          3,
        ],
        "direction": "ENE",
      },
    },
    "1586991600": Object {
      "avgWeather": Object {
        "description": "clear sky",
        "icon": "04d",
      },
      "descriptions": Array [
        "clear sky",
        "clear sky",
        "clear sky",
        "broken clouds",
        "scattered clouds",
        "broken clouds",
        "overcast clouds",
        "overcast clouds",
      ],
      "displayName": "Today",
      "humidity": Object {
        "avg": "80.00%",
        "data": Array [
          78,
          86,
          91,
          81,
          72,
          72,
          76,
          84,
        ],
      },
      "icons": Array [
        "01n",
        "01n",
        "01d",
        "04d",
        "03d",
        "04d",
        "04d",
        "04n",
      ],
      "rain": Object {
        "avg": "0.00mm",
        "data": Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
      },
      "snow": Object {
        "avg": "0.00mm",
        "data": Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
      },
      "temps": Object {
        "avg": "9°C",
        "data": Array [
          7,
          7,
          6,
          10,
          11,
          11,
          10,
          8,
        ],
        "max": "11°C",
        "min": "6°C",
      },
      "times": Array [
        1586995200,
        1587006000,
        1587016800,
        1587027600,
        1587038400,
        1587049200,
        1587060000,
        1587070800,
      ],
      "wind": Object {
        "avg": "3.71m/s",
        "data": Array [
          3,
          4,
          4,
          4,
          4,
          4,
          4,
          3,
        ],
        "direction": "NE",
      },
    },
    "1587078000": Object {
      "avgWeather": Object {
        "description": "light rain",
        "icon": "10d",
      },
      "descriptions": Array [
        "broken clouds",
        "scattered clouds",
        "broken clouds",
        "light rain",
        "light rain",
        "broken clouds",
        "light rain",
        "light rain",
      ],
      "displayName": "Tomorrow",
      "humidity": Object {
        "avg": "83.88%",
        "data": Array [
          86,
          89,
          86,
          82,
          81,
          75,
          81,
          91,
        ],
      },
      "icons": Array [
        "04n",
        "03n",
        "04d",
        "10d",
        "10d",
        "04d",
        "10d",
        "10n",
      ],
      "rain": Object {
        "avg": "0.27mm",
        "data": Array [
          0,
          0,
          0,
          0.22,
          0.56,
          0,
          0.51,
          0.88,
        ],
      },
      "snow": Object {
        "avg": "0.03mm",
        "data": Array [
          0,
          0,
          0,
          0,
          0,
          0.22,
          0,
          0,
        ],
      },
      "temps": Object {
        "avg": "10°C",
        "data": Array [
          9,
          9,
          9,
          10,
          9,
          11,
          10,
          10,
        ],
        "max": "11°C",
        "min": "9°C",
      },
      "times": Array [
        1587081600,
        1587092400,
        1587103200,
        1587114000,
        1587124800,
        1587135600,
        1587146400,
        1587157200,
      ],
      "wind": Object {
        "avg": "5.66m/s",
        "data": Array [
          4,
          5,
          6,
          5,
          6,
          7,
          7,
          6,
        ],
        "direction": "E",
      },
    },
    "1587164400": Object {
      "avgWeather": Object {
        "description": "overcast clouds",
        "icon": "04d",
      },
      "descriptions": Array [
        "light rain",
        "overcast clouds",
        "overcast clouds",
        "overcast clouds",
        "overcast clouds",
        "overcast clouds",
        "overcast clouds",
        "overcast clouds",
      ],
      "displayName": null,
      "humidity": Object {
        "avg": "85.25%",
        "data": Array [
          92,
          90,
          90,
          85,
          78,
          78,
          82,
          87,
        ],
      },
      "icons": Array [
        "10n",
        "04n",
        "04d",
        "04d",
        "04d",
        "04d",
        "04d",
        "04n",
      ],
      "rain": Object {
        "avg": "0.09mm",
        "data": Array [
          0.7,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
      },
      "snow": Object {
        "avg": "0.00mm",
        "data": Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
      },
      "temps": Object {
        "avg": "9°C",
        "data": Array [
          10,
          9,
          8,
          9,
          10,
          10,
          10,
          9,
        ],
        "max": "10°C",
        "min": "8°C",
      },
      "times": Array [
        1587168000,
        1587178800,
        1587189600,
        1587200400,
        1587211200,
        1587222000,
        1587232800,
        1587243600,
      ],
      "wind": Object {
        "avg": "5.36m/s",
        "data": Array [
          6,
          6,
          5,
          6,
          6,
          5,
          4,
          4,
        ],
        "direction": "E",
      },
    },
    "1587250800": Object {
      "avgWeather": Object {
        "description": "light rain",
        "icon": "04d",
      },
      "descriptions": Array [
        "light rain",
        "light rain",
        "light rain",
        "overcast clouds",
        "broken clouds",
        "scattered clouds",
        "broken clouds",
        "overcast clouds",
      ],
      "displayName": null,
      "humidity": Object {
        "avg": "77.13%",
        "data": Array [
          86,
          83,
          82,
          78,
          66,
          68,
          73,
          81,
        ],
      },
      "icons": Array [
        "10n",
        "10n",
        "10d",
        "04d",
        "04d",
        "03d",
        "04d",
        "04n",
      ],
      "rain": Object {
        "avg": "0.17mm",
        "data": Array [
          0.24,
          0.91,
          0.23,
          0,
          0,
          0,
          0,
          0,
        ],
      },
      "snow": Object {
        "avg": "0.00mm",
        "data": Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
      },
      "temps": Object {
        "avg": "9°C",
        "data": Array [
          9,
          9,
          8,
          9,
          11,
          11,
          10,
          9,
        ],
        "max": "11°C",
        "min": "8°C",
      },
      "times": Array [
        1587254400,
        1587265200,
        1587276000,
        1587286800,
        1587297600,
        1587308400,
        1587319200,
        1587330000,
      ],
      "wind": Object {
        "avg": "4.78m/s",
        "data": Array [
          5,
          5,
          5,
          5,
          6,
          5,
          4,
          4,
        ],
        "direction": "E",
      },
    },
    "1587337200": Object {
      "avgWeather": Object {
        "description": "light rain",
        "icon": "10d",
      },
      "descriptions": Array [
        "light rain",
        "light rain",
        "light rain",
        "light rain",
        "light rain",
        "light rain",
      ],
      "displayName": null,
      "humidity": Object {
        "avg": "86.00%",
        "data": Array [
          87,
          82,
          87,
          88,
          86,
          86,
        ],
      },
      "icons": Array [
        "10n",
        "10n",
        "10d",
        "10d",
        "10d",
        "10d",
      ],
      "rain": Object {
        "avg": "1.07mm",
        "data": Array [
          0.21,
          0.72,
          1.81,
          1.52,
          0.89,
          1.3,
        ],
      },
      "snow": Object {
        "avg": "0.00mm",
        "data": Array [
          0,
          0,
          0,
          0,
          0,
          0,
        ],
      },
      "temps": Object {
        "avg": "8°C",
        "data": Array [
          9,
          8,
          7,
          7,
          8,
          8,
        ],
        "max": "9°C",
        "min": "7°C",
      },
      "times": Array [
        1587340800,
        1587351600,
        1587362400,
        1587373200,
        1587384000,
        1587394800,
      ],
      "wind": Object {
        "avg": "6.92m/s",
        "data": Array [
          6,
          6,
          7,
          9,
          8,
          5,
        ],
        "direction": "ENE",
      },
    },
  },
  "times": Object {
    "sunrise": 1586928264,
    "sunset": 1586978702,
  },
}
`;
